src/types.ts

```diff
@@ -0,0 +1,20 @@
+export interface Widget {
+  id: string;
+  name: string;
+  description: string;
+}
+
+export interface WidgetsResponse {
+  widgets: Widget[];
+}
+
+
+export interface Gadget {
+  id: string;
+  name: string;
+  description: string;
+}
+
+export interface GadgetsResponse {
+    gadgets: Gadget[];
+  }
```

Comment:
The rest of this looks great, but it seems the indentation is a little funky here.

We should probably create a task to add a prettier auto formatting commit hook. Not needed in this PR, but something we want to track.

Reply:
Great catch. Fixed.